.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,29 @@
+name: build
+on: [push]
+jobs:
+  build_and_push:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Extract branch name
+        run: echo "normalized_branch_name=$(stripped=${GITHUB_REF#refs/heads/} && echo ${stripped/\//-})" >> $GITHUB_ENV
```

Comment:
What does GITHUB_ENV evaluate to here?

Reply:
It's a built in: https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable - anything you append to `$GITHUB_ENV` is then available as an environment variable in further steps.

Reply:
Ah, interesting, I didn't know about that. I've been saving things as outputs to handle this in other workflows